packages/types/src/type/ExtrinsicEra.ts

```diff
@@ -2,33 +2,166 @@
+    if (this.index === 0 ) {
+      return super.toU8a()
+    } else {
+      return (this.asMortalEra as MortalEra).toU8a(isBare);
+    }
+  }
+}
+
+export class ImmortalEra extends U8a {
+  constructor(value?: AnyU8a) {
+    super(value);
+  }
+}
+
+export type MortalEraValue = [U8, U8];
 
 /**
- * @name ExtrinsicEra
+ * @name MortalEra
  * @description
- * The era for an extrinsic, indicating either a mortal or immortal extrinsic
+ * The MortalEra for an extrinsic, indicating period and phase
  */
-export default class ExtrinsicEra extends U8a {
-  constructor (value?: AnyU8a) {
-    super(
-      ExtrinsicEra.decodeExtrinsicEra(value)
-    );
+export class MortalEra extends Tuple {
+  constructor (value?: any) {
+    super({
+      period : U8, phase : U8
+    }, MortalEra.decodeMortalEra(value));
+  }
+
+  private static decodeMortalEra (value: EraMethod | Uint8Array | string): MortalEraValue {
+    if (isHex(value)) {
+      return MortalEra.decodeMortalEra(hexToU8a(value.toString()));
+    } else if (isU8a(value)) {
+      const first = u8aToBn(value.subarray(0, 1)).toNumber();
+      let second = u8aToBn(value.subarray(1, 2)).toNumber();
+      const encoded: number = first + (second << 8);
+      const period = 2 << (encoded % (1 << 4));
+      const quantizeFactor = Math.max(period >> 12, 1);
+      let phase = (encoded >> 4) * quantizeFactor;
+      if (period >= 4 && phase < period) {
+        return [new U8(period), new U8(phase)];
+      }
+      throw new Error('Invalid data passed to Mortal era');
+    } else if (isObject(value)) {
+        const {current} = value;
+        const {period} = value;
+        let calPeriod = Math.pow(2, Math.ceil(Math.log2(period.toNumber())));
+        calPeriod = Math.min( Math.max(calPeriod, 4), 1<< 16);
+        const phase = current.toNumber() % calPeriod;
+        const factor = 12;
+        const quantizeFactor = calPeriod >> factor > 1 ? calPeriod >> factor : 1;
+        let quantizedPhase = phase / quantizeFactor * quantizeFactor;
+        return [new U8(calPeriod), new U8(quantizedPhase)];
+    }
+    return [new U8(), new U8()];
+  }
+  /**
+   * @description The justification [[U64]]
+   */
+  get period (): U8 {
+    return this[0] as U8;
+  }
+
+  /**
+   * @description The round this justification wraps as a [[U64]]
+   */
+  get phase (): U8 {
+    return this[1] as U8;
   }
 
-  static decodeExtrinsicEra (value?: AnyU8a): Uint8Array {
-    if (value) {
-      const u8a = u8aToU8a(value);
+  /**
+   * @description Encodes the value as a Uint8Array as per the parity-codec specifications
+   * @param isBare true when the value has none of the type-specific prefixes (internal)
+   */
+  toU8a (isBare?: boolean): Uint8Array {
 
-      // If we have a zero byte, it is immortal (1 byte in length), otherwise we have
-      // the era details following as another byte
-      return u8a.subarray(0, (u8a[0] === 0) ? 1 : 2);
+    const period = this.period;
```

Comment:
this assignment and the next line are tedious. or you want to change it to `const period = this.period.toNumber()` which will make it more useful